harness/suite.go

```diff
@@ -203,6 +208,26 @@ func (s *Suite) Run() (err error) {
 		return err
 	}
 
+	for _, reporter := range s.reporters {
```

Comment:
We're looping through 'reporters' multiple times, and it feels rather fragile.

I think encapsulating a slice of reporters as its own type (`type Reporters []Reporter`) and then defining methods on it (like 'Create` and `WriteOutput` and such). I think that would make all of this code cleaner.